core/src/main/java/org/apache/accumulo/core/spi/compaction/DefaultCompactionPlanner.java

```diff
@@ -216,7 +217,9 @@ public void init(InitParameters params) {
               "'numThreads' should not be specified for external compactions");
           String group = Objects.requireNonNull(executorConfig.group,
               "'group' must be specified for external type");
-          ceid = params.getExecutorManager().getExternalExecutor(group);
+
+          ceid =
+              params.getExecutorManager().getExternalExecutor(serviceId.canonical() + "." + group);
```

Comment:
This is slightly different than what I am thinking and this may still allow for a graph instead of tree.  I was thinking of something like the following

A compactor must be started with a group name of the form <service name>.<queue name> for its group.  So starting a compactor could look like `accumulo compactor - compactor.group=system.default`.  The compactor could check the group is of this form, it could even check if the compaction service is actually configured.

The config specifies either the group or queue like the following (excluded some things in the config that could be dropped.

```
tserver.compaction.major.service.system.planner.opts.executors=[{'group':'system.default'}]
```
or 


```
tserver.compaction.major.service.system.planner.opts.executors=[{'queue':'default'}]
```

When the call is made to `params.getExecutorManager().getExternalExecutor(X)` X really only needs to be the queue name.  The implementation of getExternalExecutor() should know the service its being called for and could append it to create the id.   If X is of the form <service>.<queue> then the impl could throw an illegal argument exception if the service portion is anything other than the expected service.

Reply:
I edited my comment a few times because `<abc>` like things were being hidden because I did not quote them in markdown.

Reply:
Ok, that makes sense. 

My goal was to support the queue only format so that way the service isn't mis-typed by the user. 
```
tserver.compaction.major.service.system.planner.opts.executors=[{'queue':'default'}]
```